src/components/text-editor/text-editor.tsx

```diff
@@ -0,0 +1,98 @@
+import { Component, Element, Prop, forceUpdate, h } from '@stencil/core';
+import { Editor, EditorProps } from '@toast-ui/react-editor';
+import { EditorType } from '@toast-ui/editor/types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import '@toast-ui/editor/dist/toastui-editor.css';
+
+/**
+ * The `limel-text-editor` component wraps the toast-ui react editor. This editor
+ * allows a rich text editing experience with markdown support.
+ */
+@Component({
+    tag: 'limel-text-editor',
+    shadow: true,
+    styleUrl: 'text-editor.scss',
+})
+export class TextEditor {
+    @Prop()
+    initialValue: string = '';
+
+    @Prop()
+    editType: EditorType = 'wysiwyg';
+
+    @Prop()
+    theme: 'dark' | 'light' = 'light';
```

Comment:
I don't think our other components have properties for light/dark mode, we should make this component work the same way